src/braket/pennylane_plugin/braket_device.py

```diff
@@ -675,6 +675,69 @@ def _run_snapshots(self, snapshot_circuits, n_qubits, mapped_wires):
 
         return outcomes
 
+    def _validate_parametrized_evolution(self, ev):
+        """Validates pulse input (ParametrizedEvolution) before converting to a PulseGate"""
+
+        # note: the pulse upload on the braket side immediately checks that the max amplitude is not exceeded,
+        # so that check has not been included here
+
+        # confirm all frequency and phase values are constant for the duration of a pulse
+        callable_freqs = [pulse.frequency for pulse in ev.H.pulses if callable(pulse.frequency)]
+        callable_phase = [pulse.phase for pulse in ev.H.pulses if callable(pulse.phase)]
+
+        if callable_freqs:
+            raise RuntimeError(f"Expected all frequencies to be constants but recieved callable(s)")
+        if callable_phase:
+            raise RuntimeError(f"Expected all phases to be constants but recieved callable(s)")
+
+        # confirm all frequencies are within permitted difference from center frequency
+        #ToDo: why does 'PERMITTED_FREQUENCY_DIFFERENCE' return 1.0 for OQC device?
+        # Rigetti returns a reasonable 400MHz - will use this for now
```

Comment:
Should be 1GHz for OQC, gonna be difficult handling different devices when they use different conventions 😬